agent/api/ecsclient/client.go

```diff
@@ -205,42 +261,21 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 		StringSetValue: utils.Uint16SliceToStringSlice(client.config.ReservedPortsUDP),
 	}
 
-	resources := []*ecs.Resource{&cpuResource, &memResource, &portResource, &udpPortResource}
-	registerRequest.TotalResources = resources
-
-	resp, err := client.standardClient.RegisterContainerInstance(&registerRequest)
-	if err != nil {
-		seelog.Errorf("Could not register: %v", err)
-		return "", err
-	}
-	seelog.Info("Registered container instance with cluster!")
-	err = validateRegisteredAttributes(registerRequest.Attributes, resp.ContainerInstance.Attributes)
-	return aws.StringValue(resp.ContainerInstance.ContainerInstanceArn), err
+	return []*ecs.Resource{&cpuResource, &memResource, &portResource, &udpPortResource}, nil
```

Comment:
The diff viewer is confused by the new functions you added here. I am assuming you just refactored parent function to getResources and the code is the same except parameterization. If not, let's talk offline.